libs/src/sdk/api/playlists/PlaylistsApi.ts

```diff
@@ -263,26 +270,78 @@ export class PlaylistsApi extends GeneratedPlaylistsApi {
       PublishPlaylistSchema
     )(requestParameters)
 
-    // Fetch playlist
-    const playlistResponse = await this.getPlaylist({
-      playlistId: requestParameters.playlistId,
-      userId: requestParameters.userId
-    })
-    const playlist = playlistResponse.data?.[0]
+    return await this.fetchAndUpdatePlaylist(
+      {
+        userId: requestParameters.userId,
+        playlistId: requestParameters.playlistId,
+        updateMetadata: (playlist) => ({
+          ...playlist,
+          isPrivate: false
+        })
+      },
+      writeOptions
+    )
+  }
 
-    if (!playlist) {
-      throw new Error(
-        `Could not fetch playlist: ${requestParameters.playlistId}`
-      )
-    }
+  /**
+   * Add a single track to the end of a playlist
+   * For more control use updatePlaylist
+   */
+  async addTrackToPlaylist(
+    requestParameters: AddTrackToPlaylistRequest,
+    writeOptions?: WriteOptions
+  ) {
+    // Parse inputs
+    parseRequestParameters(
+      'addTrackToPlaylist',
+      AddTrackToPlaylistSchema
+    )(requestParameters)
 
-    return await this.updatePlaylist(
+    const currentBlock = await this.entityManager.getCurrentBlock()
+
+    return await this.fetchAndUpdatePlaylist(
       {
         userId: requestParameters.userId,
         playlistId: requestParameters.playlistId,
-        metadata: {
+        updateMetadata: (playlist) => ({
           ...playlist,
-          isPrivate: false
+          playlistContents: [
+            ...playlist.playlistContents,
+            {
+              trackId: requestParameters.trackId,
+              timestamp: currentBlock.timestamp
+            }
+          ]
+        })
+      },
+      writeOptions
+    )
+  }
+
+  /**
+   * Removes a single track at the given index of playlist
+   * For more control use updatePlaylist
+   */
+  async removeTrackFromPlaylist(
+    requestParameters: RemoveTrackFromPlaylistRequest,
+    writeOptions?: WriteOptions
+  ) {
+    // Parse inputs
+    const { trackIndex } = parseRequestParameters(
+      'removeTrackFromPlaylist',
+      RemoveTrackFromPlaylistSchema
+    )(requestParameters)
+
+    return await this.fetchAndUpdatePlaylist(
+      {
+        userId: requestParameters.userId,
+        playlistId: requestParameters.playlistId,
+        updateMetadata: (playlist) => {
+          playlist.playlistContents.splice(trackIndex, 1)
```

Comment:
wonder if we should add a check to make sure the track index exists

Reply:
Oh yes, good idea